shopinvader_auth_jwt/models/shopinvader_backend.py

```diff
@@ -18,12 +18,28 @@ class ShopinvaderBackend(models.Model):
 
     @api.model
     def _get_from_jwt_aud(self, aud):
+        """On a record, ensure aud is valid
+        on an empty recordset, will retrieve a record
+        with aud"""
         if not aud:
             return self.browse([])
         if isinstance(aud, str):
             aud = [aud]
-        backends = self.search([("jwt_aud", "in", aud)])
-        if len(backends) != 1:
-            _logger.warning("%d backends found for JWT aud %r", aud)
-            return self.browse([])
-        return backends
+        if len(self.ids > 0):
+            # self is a recordset
+            if self.jwt_aud in aud:
```

Comment:
this line will fail if `len(self.ids) > 1` We should maybe raise an exception in such a case.

Reply:
fixed with self.ensure_one()